### ISSUE TITLE
InputNumber: Form does not submit on enter

### ISSUE THREAD

**Issue by sane6** [openvi-foundation/openvue#573]
### Describe the bug

Usually you can submit forms via `enter` on inputs.
This works for default `<input type="text">` as well as your `InputText` component - but it does not work for `InputNumber`

You can check the example link below, go to an input, press `enter` and see the alert "form submitted"
Works for InputText but not for InputNumber

### Reproducer

https://codesandbox.io/p/sandbox/primevue-demo-forked-mkztsw?file=%2Fsrc%2FApp.vue%3A8%2C38

### PrimeVue version

3.30

### Vue version

3.x

### Language

ALL

### Build / Runtime

Vue CLI App

### Steps to reproduce the behavior

1. go to https://codesandbox.io/p/sandbox/primevue-demo-forked-mkztsw?file=%2Fsrc%2FApp.vue%3A29%2C10-29%2C24
2. go to the inputnumer component
3. press enter
4. see no form submit is triggered

### Expected behavior

form should be submitted

Copied to track in our fork; not affiliated with the original report being closed/resolved upstream

Originally reported at: [primefaces#4199](https://github.com/primefaces/primevue/issues/4199)
Originally reported by: https://github.com/Anubarak
